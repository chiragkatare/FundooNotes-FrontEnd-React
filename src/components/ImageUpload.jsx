import React from 'react';

export default class ImageUpload extends React.Component {

    state = {
        open: false,
        messageInfo: {},
    };

    handleImageClick = () => {
        this.fileInput.click();
    }

    handleNotePic = (event) => {

        this.props.handleImageUpload(event.target.files[0]);
    }

    render() {
        return (
            <div>
                <div className='note-icon-div' role='Button' onClick={this.handleImageClick} >
                    <img src={require('../assets/icons/AddImage.svg')} alt="" />
                </div>
                <input

                    style={{ display: 'none' }}
                    accept='image/*'
                    ref={fileInput => this.fileInput = fileInput}
                    type='file'
                    name='profiepic'
                    placeholder='Change Profile Pic'
                    onChange={this.handleNotePic}
                />
            </div>
        );
    }
}



