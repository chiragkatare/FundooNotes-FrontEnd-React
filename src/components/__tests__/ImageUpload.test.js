import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, configure } from 'enzyme';
import ImageUpload from '../ImageUpload';

const mockFn = jest.fn();


// configure({ adapter: new Adapter() });




let props = {
    name: 'password',
    type: 'text',
    placeholder: 'hello',
    label: 'label',
    onChange: mockFn,
    handleImageUpload:mockFn

}

const component = shallow(<ImageUpload {...props} />);

describe('Input', () => {
    it('should render correctly in "debug" mode', () => {

        // const component = mount(<Login debug />);
        expect(component).toMatchSnapshot();
    });

    it('should render correctly in with props', () => {


        let textfield = component.find('input');
        textfield.simulate('change', {
            target: {
                value: 8,
                files:["sd"]
            }
        })
    })
});