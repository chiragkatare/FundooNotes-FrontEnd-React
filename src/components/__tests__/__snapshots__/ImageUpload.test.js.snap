// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly in "debug" mode 1`] = `
<div>
  <div
    className="note-icon-div"
    onClick={[Function]}
    role="Button"
  >
    <img
      alt=""
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "AddImage.svg",
        }
      }
    />
  </div>
  <input
    accept="image/*"
    name="profiepic"
    onChange={[Function]}
    placeholder="Change Profile Pic"
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
